editor/blocks/quote-block/index.js

```diff
@@ -0,0 +1,23 @@
+const { query, html, text } = wp.blocks.query;
+
+wp.blocks.registerBlock( 'wp/quote', {
+	title: 'Quote',
+	icon: 'format-quote',
+
+	attributes: {
+		value: query( 'blockquote', html( 'p' ) ),
+		citation: query( 'cite,footer', text() )
```

Comment:
Here's some info: https://w3c.github.io/html/grouping-content.html#the-blockquote-element
There are lots of possibilities to format this.